### PR TITLE
Added rook ceph operator config map reconcile func

### PR DESCRIPTION
**Pull request by ezio-auditore**
Reconcile function to edit `rook-ceph-operator-config` to disable these flags in DataFoundation service

- ROOK_CSI_ENABLE_CEPHFS
- ROOK_CSI_ENABLE_RBD

### CODE REVIEW

File: controllers/datafoundation.go [hunk 1]
```diff
@@ -114,3 +119,29 @@ func (r *dataFoundationReconciler) reconcileOnboardingValidationSecret() error {
 	}
 	return nil
 }
+
+func (r *dataFoundationReconciler) reconcileRookCephOperatorConfig() error {
+	rookConfigMap := &corev1.ConfigMap{}
+	rookConfigMap.Name = "rook-ceph-operator-config"
+	rookConfigMap.Namespace = r.managedFusionOffering.Namespace
```
Comment (nb-ohad) [inline controllers/datafoundation.go:126]: Why do we have some reconciled resources defined on the reconciler (and identified in `initReconciler`) and some defined  in the reconciler function?

**Review by nb-ohad — CHANGES_REQUESTED** [review]

**Comment by nb-ohad** [comment]
@ezio-auditore Please add a description to the PR

File: controllers/datafoundation.go (hunk 1 above)
Comment (ezio-auditore) [reply controllers/datafoundation.go]: Updated

**Review by nb-ohad — DISMISSED** [review]

File: controllers/datafoundation.go [hunk 2]
```diff
@@ -83,6 +89,9 @@ func (r *dataFoundationReconciler) initReconciler(offeringReconciler *ManagedFus
 	r.cephIngressNetworkPolicy = &netv1.NetworkPolicy{}
 	r.cephIngressNetworkPolicy.Name = "ceph-ingress-rule"
 	r.cephIngressNetworkPolicy.Namespace = r.Namespace
+	r.rookConfigMap = &corev1.ConfigMap{}
```
Comment (nb-ohad) [inline controllers/datafoundation.go:92]:
```suggestion
	
	r.rookConfigMap = &corev1.ConfigMap{}
```

File: controllers/datafoundation.go [hunk 3]
```diff
@@ -221,3 +231,25 @@ func (r *dataFoundationReconciler) reconcileCephIngressNetworkPolicy() error {
 	}
 	return nil
 }
+
+func (r *dataFoundationReconciler) reconcileRookCephOperatorConfig() error {
+	if err := r.get(r.rookConfigMap); err != nil {
+		// Because resource limits will not be set, failure to get the Rook ConfigMap results in failure to reconcile.
```
Comment (nb-ohad) [inline controllers/datafoundation.go:237]: This is an error about resource limits, but I don't see any code that update the resource limits

**Review by nb-ohad — CHANGES_REQUESTED** [review]

**Review by nb-ohad — APPROVED** [review]

**Pull request merged**